src/plugins/data_source/server/data_source_service.ts

```diff
@@ -6,11 +6,13 @@
 import { Logger, OpenSearchClient, SavedObjectsClientContract } from '../../../../src/core/server';
 import { DataSourcePluginConfigType } from '../config';
 import { OpenSearchClientPool, configureClient } from './client';
+import { CryptographyClient } from './cryptography';
 export interface DataSourceServiceSetup {
   getDataSourceClient: (
     dataSourceId: string,
     // this saved objects client is used to fetch data source on behalf of users, caller should pass scoped saved objects client
-    savedObjects: SavedObjectsClientContract
+    savedObjects: SavedObjectsClientContract,
```

Comment:
```
    // this saved objects client is used to fetch data source on behalf of users, caller should pass scoped saved objects client
```
Can I ask more context on this?

Reply:
it should be scoped client, which means a client tight to a user in the request. Different user may have access to different saved object, including datasource or credentials.

Reply:
> it should be scoped client, which means a client tight to a user in the request. Different user may have access to different saved object, including datasource or credentials.

Follow up question - Is this already possible with the current scoped clienet? What is the isolation based on?